### PR TITLE
O3-1281 Visit Note - Add Ability to Mark Primary and Secondary Diagnoses

### PR DESCRIPTION
**Pull request by abertnamanya**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [ ] My work includes tests or is validated by existing tests.

## Summary

1. Form allows selection of multiple primary diagnosis and multiple secondary diagnosis 
2. Plus diagnosis could be detached/removed from respective category 
3. After submission diagnoses are displayed in the Visit Note data table

## Screenshots

https://user-images.githubusercontent.com/19745793/176147752-e0b93bd2-b803-42a7-8039-6a19f731c6bd.mp4

## Related Issue

https://issues.openmrs.org/browse/O3-1281

### CODE REVIEW

**Comment by abertnamanya** [comment]
Hi @ZacButko Here is the PR, But I did skip the tests never wanted to delay this ticket could you find time and resolve the tests for the Visit Note.
Thank you.

File: packages/esm-patient-notes-app/src/types/index.ts [hunk 1]
```diff
@@ -32,29 +33,33 @@ export interface PatientNote {
 }
 
 export interface DiagnosisData {
-  word: null;
-  conceptName: {
-    id: number;
+  uuid: string;
+  display: string;
+  name: {
```
Comment (ibacher) [inline packages/esm-patient-notes-app/src/types/index.ts:38]: Do we care about the name property at all? We only really need to include the properties we care about (we should use `display`).

File: packages/esm-patient-notes-app/src/types/index.ts [hunk 2]
```diff
@@ -32,29 +33,33 @@ export interface PatientNote {
...
     uuid: string;
-    conceptMappings: Array<ConceptMapping>;
-    preferredName: string;
+    name: string;
+    description: string;
   };
+  names: Array<conceptNames>;
+  mappings: Array<ConceptMapping>;
+}
+export interface conceptNames {
```
Comment (ibacher) [inline packages/esm-patient-notes-app/src/types/index.ts:51]: Should be capitalised, but again, I don't think this is useful information to maintain...

```suggestion
export interface ConceptNames {
```

File: packages/esm-patient-notes-app/src/notes/visit-notes.resource.ts [hunk 3]
```diff
@@ -106,7 +105,16 @@ export function saveVisitNote(abortController: AbortController, payload: VisitNo
...
+      'Content-Type': 'application/json',
+    },
+    method: 'POST',
+    body: payload,
+  });
+}
 
 function getConceptReferenceTermCode(conceptMapping: Array<ConceptMapping>): ConceptMapping {
-  return conceptMapping.find((concept) => concept.conceptReferenceTerm.conceptSource.name === 'ICD-10-WHO');
+  return conceptMapping.find((concept) => concept.conceptReferenceTerm.display);
```
Comment (ibacher) [inline packages/esm-patient-notes-app/src/notes/visit-notes.resource.ts:119]: What is this supposed to do?

File: packages/esm-patient-notes-app/src/notes/visit-notes.resource.ts [hunk 4]
```diff
@@ -89,10 +87,11 @@ function formatDiagnoses(diagnoses: Array<DiagnosisData>): Array<Diagnosis> {
...
 function mapDiagnosisProperties(diagnosis: DiagnosisData): Diagnosis {
   return {
-    concept: diagnosis.concept,
-    conceptReferenceTermCode: getConceptReferenceTermCode(diagnosis.concept.conceptMappings).conceptReferenceTerm.code,
+    uuid: diagnosis.uuid,
+    concept: diagnosis.name,
+    conceptReferenceTermCode: getConceptReferenceTermCode(diagnosis.mappings).conceptReferenceTerm.display,
     primary: false,
-    confirmed: false,
+    confirmed: true,
```
Comment (ibacher) [inline packages/esm-patient-notes-app/src/notes/visit-notes.resource.ts:94]: Why do we default to assuming a diagnosis is confirmed? ("Confirmed" in this context generally means "we have a lab test that proves the patient has x")

**Review by ibacher — CHANGES_REQUESTED** [review]
@abertnamanya Thanks for this! Obviously, we need to fix the tests before this can be merged and there are one or two minor points I've raised in comments below.

**Comment by hadijahkyampeire** [comment]
@abertnamanya are you still available to work on this or you would prefer someone to take it on?

**Comment by abertnamanya** [comment]
> @abertnamanya are you still available to work on this or you would prefer someone to take it on?

Dear @hadijahkyampeire, this was taken over by @icrc-loliveira in this PR https://github.com/openmrs/openmrs-esm-patient-chart/pull/757
**Regards**,

**Comment by hadijahkyampeire** [comment]
Oh that’s okay, then we should close this PR since it’s work is already merged.

**Pull request closed without merging**